src/Apple/Provider.php

```diff
@@ -114,43 +114,53 @@ protected function getUserByToken($token)
 
     protected function getClientSecret()
     {
-        if (!$this->jwtConfig) {
-            $this->getJwtConfig(); // Generate Client Secret from private key if not set.
+        if (!empty($this->privateKey)) {
+            $this->clientSecret = $this->generateApplePrivateTokenString();
+            config()->set('services.apple.client_secret', $this->clientSecret);
         }
 
         return $this->clientSecret;
     }
 
-    protected function getJwtConfig()
+    protected function createJwtConfig(): void
     {
-        if (!$this->jwtConfig) {
+        if (!$this->jwtConfig instanceof Configuration) {
             $private_key_path = $this->getConfig('private_key', '');
             $private_key_passphrase = $this->getConfig('passphrase', '');
-            $signer = $this->getConfig('signer', '');
+            $signerClassName = $this->getConfig('signer', '');
 
-            if (empty($signer) || !class_exists($signer)) {
-                $signer = !empty($private_key_path) ? \Lcobucci\JWT\Signer\Ecdsa\Sha256::class : AppleSignerNone::class;
+            if (empty($signerClassName) || !class_exists($signerClassName) || !is_a($signerClassName, Signer::class, true)) {
+                $signerClassName = EcdsaSha256::class;
```

Comment:
The `AppleSignerNone` is gone.

I don't believe there is a use case for a configurable signer class, as the only signature allowed by Apple is ES256 right now, but I added some sanity check beyond "class exists".